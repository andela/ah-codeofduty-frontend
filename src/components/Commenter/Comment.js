import React from 'react';
import PropTypes from 'prop-types';
import {timeConverter} from '../../utils/timeConverter'

const propTypes = {};

const defaultProps = {};

export default class Comment extends React.Component {
    constructor(props) {
        super(props);
        this.state = {body: ''};
    }

    onChange = (event) => {
        this.setState({ body: event.target.value })
    }

    onSubmit = () => {
        const { comment, reply, slug } = this.props;
        reply(slug, comment.id, this.state);
    }

    render() {
        const { comment } = this.props;
        return (
            <React.Fragment>

            <div class="p-3 mb-2">
                <hr/>
                <img class="avatar" src="../photos/profile.jpg"></img>
                <span class="pl-3 font-satisfy"><b>@commenter</b></span>
                <p class="pt-3">
                    {comment.body}
                </p>
                <p>Modified: <i>{timeConverter(comment.updated_at)}</i></p>

                <input class="" type="text" placeholder="Your Reply ..." onChange={this.onChange}></input>
                <button onClick={this.onSubmit} class="btn btn-warning ml-3 mt-2 p-2 px-2">Reply<i class="ml-2 fa fa-paper-plane"
                                    aria-hidden="true"></i></button>
                <div style={{paddingRight: '100px'}}>
                    {comment.thread.map(response => 
                    <React.Fragment>
                    <div class="ml-2 container mt-3 border-left border-warning">
                    <img class="small-avatar" src="../photos/profile.jpg"></img>
                    <span class="pr-2 font-satisfy"><b>@replier <span class="fa fa-angle-double-right"></span></b></span>
                    <span class="pt-3">{response.body}</span>
                    <br/>
                    <p>Modified: <i>{timeConverter(response.updated_at)}</i></p>
                    <br/>
                    </div>
                    </React.Fragment>)}
                </div>
                <hr/>
            </div>
            </React.Fragment>
        );
    }
}

 Comment.propTypes = propTypes;
 Comment.defaultProps = defaultProps;
