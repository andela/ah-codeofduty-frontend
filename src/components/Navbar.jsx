import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import authUser from '../utils/authUser';
import LoginForm from './Authentication/LoginForm/LoginForm';

class Navbar extends Component {
  render() {
    return (
      <div className="App">
        <nav>
          <ul>
            <li className="App-Logo">Authors' Haven</li>
            <li>
              <Link to="/">Home</Link>
            </li>
            {authUser()
              ? (
                <li>
                <a onClick={() => localStorage.removeItem('user')} href="#">Logout</a>
              </li>
              ) : (
                <li>
                <a data-toggle="modal" data-target="#loginModal">Login</a>
              </li>
              )
      }
          </ul>
        </nav>
        <hr className="HR" />
        <LoginForm />
      </div>
    );
  }
}

export default Navbar;
