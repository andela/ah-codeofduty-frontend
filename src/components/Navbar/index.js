import React, { Component } from 'react';
import { Link } from 'react-router-dom';
// import Index from "./components/Home";
// import Login from "./components/Login";
// import Profile from "./components/Profile/Profile";

class Home extends Component {
  render() {
    return (
      <div className="App">
        <nav>
          <ul>
            <li className="App-Logo">Authors' Haven</li>
            <li>
              <Link to="/">Home</Link>
            </li>
            <li>
              <Link to="/login/">Login</Link>
            </li>
            <li>
              <Link to="/profile/">Profile</Link>
            </li>
          </ul>
        </nav>
        <hr className="HR" />

        {/* <Route exact path="/" component={Index} /> */}
        {/* <Route path="/login/" component={Login} />
          <Route path="/profile" component={Profile} /> */}
      </div>
    );
  }
}

export default Home;
