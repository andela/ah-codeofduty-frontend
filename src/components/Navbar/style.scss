/*Custom Home Navbar*/
html,
body {
  height: 100%;
}
navbar {
  padding: 0px !important;
  -webkit-box-shadow: 4px 9px 25px -6px rgba(77, 77, 77, 0.61);
  -moz-box-shadow: 4px 9px 25px -6px rgba(77, 77, 77, 0.61);
  box-shadow: 4px 9px 25px -6px rgba(77, 77, 77, 0.61);
}

.logo {
  border-right: 2px solid orange;
  color: orange;
}

.logo:hover {
  color: black;
  opacity: 0.8;
  text-decoration: none;
}

.badges {
  position: relative;
  top: -20px;
  left: -25px;
  font-size: 12px;
}

/*Side by side*/
.login-form {
  width: 340px;
  margin: 30px auto;
}
.login-form form {
  margin-bottom: 15px;
  background: #f7f7f7;
  box-shadow: 0px 1px 10px rgba(0, 0, 0, 0.3);
  padding: 30px;
}
.login-form h2 {
  margin: 0 0 15px;
}
.login-form .hint-text {
  color: #777;
  padding-bottom: 15px;
  text-align: center;
}
.form-control,
.btn {
  min-height: 38px;
  border-radius: 2px;
}
.login-btn {
  font-size: 15px;
  font-weight: bold;
}
.or-seperator {
  margin: 20px 0 10px;
  text-align: center;
  border-top: 1px solid #ccc;
}
.or-seperator i {
  padding: 0 10px;
  background: #f7f7f7;
  position: relative;
  top: -11px;
  z-index: 1;
}
.social-btn .btn {
  border-radius: 5px;
  padding-bottom: 4px;
  padding-top: 8px;
}
.social-btn .btn:hover {
  box-shadow: 2px 2px 4px #888888;
}

.social-btn .btn i {
  float: left;
  margin: 0;
}
.input-group-addon .fa {
  font-size: 18px;
}

.form-control::placeholder {
  color: rgba(73, 80, 87, 0.4);
}

.form-control:hover {
  box-shadow: 0px 1px 4px rgba(0, 0, 0, 0.3);
}

/*Authentication*/
* {
  box-sizing: border-box;
}

/* Create two equal columns that floats next to each other */
.column {
  float: left;
  width: 50%;
  padding: 10px;
  height: 300px; /* Should be removed. Only for demonstration */
}

/* Clear floats after the columns */
.row:after {
  content: "";
  display: table;
  clear: both;
}

#or_divide {
  height: 100%;
}

/*card footer*/

.block_icons {
  display: block;
}

.nav-item {
  cursor: pointer;
}
.nav-item :hover {
  opacity: 10px;
}

.navbar-brand {
  background: orange;
  padding: 10px;
}

.input-group-addon {
  padding-top: 10px;
  padding-right: 6px;
}

/*Social Logins color codes*/
#color_facebook {
  background: #3b5998;
}
#color_twitter {
  background: #1da1f2;
}
#color_google {
  background: #db4437;
}

/*post image*/
.w-100 {
  padding: 1rem !important;
}

.container,
#card_content {
  padding: 10px;
}
.fa-heart {
  color: rgb(214, 124, 116);
  opacity: 20;
}
.fa-comment {
  color: #413e3e;
  opacity: 20;
}
.fa-user {
  border: #232526;
  background: #1da1f2;
  padding: 10px;
}

.card-footer {
  margin-top: 20px;
}

/*search*/
.form-control {
  padding-left: 2.375rem;
}

.pl {
  padding-left: 2.375rem;
}

.search-input {
  display: inline-flex;
  color: #aaa;
  padding-left: 1.6rem;
}

.search {
  position: absolute;
  width: 2.375rem;
  line-height: 2.375rem;
  text-align: center;
  pointer-events: none;
  color: #aaa;
}

.dropdown-menu :hover {
  background-color: #f1f1f1;
}

.card {
  margin-bottom: 1rem !important;
}

#search {
  line-height: 2.375rem;
}

#myInput {
  margin-top: -8px;
  border: 0;
  border-radius: 0;
  background: #f1f1f1;
}

.flexed {
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  justify-content: space-evenly;
  align-items: center;
}

.fa-facebook-f:before,
.fa-facebook:before,
.fa-twitter:before,
.fa-google:before {
  content: "\F09A";
  line-height: 24px;
  font-size: 1.6rem;
}
.modal.show .modal-dialog {
  margin-top: 8em;
}

@media (max-width: 992px) {
  .search-input {
    display: inline-flex;
    padding-left: 0;
  }
}

mark {
  background-color: khaki;
}
