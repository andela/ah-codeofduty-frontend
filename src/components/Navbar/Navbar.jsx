import React, { Component } from 'react';
import { Link } from 'react-router-dom';

import 'bootstrap/dist/js/bootstrap.min';
import 'bootstrap/dist/css/bootstrap.min.css';
import './style.scss';
import '../../App.scss';

import authUser from '../../utils/authUser';
import LoginForm from '../Authentication/LoginForm/LoginForm';

class Navbar extends Component {
  render() {
    return (

      <nav
        className="navbar navbar-expand-lg navbar-light bg-light font-raleway"
        id="navbar_home"
      >
        <a
          className="logo pl-1 pr-4 mr-3 text-large font-weight-bold App-Logo"
          href="/"
        >
          Authors' Haven
        </a>
        <button
          className="navbar-toggler"
          type="button"
          data-toggle="collapse"
          data-target="#navbarTogglerDemo02"
          aria-controls="navbarTogglerDemo02"
          aria-expanded="false"
          aria-label="Toggle navigation"
        >
          <span className="navbar-toggler-icon" />
        </button>

        <div className="collapse navbar-collapse" id="navbarTogglerDemo02">
          <ul className="navbar-nav mr-auto mt-2 mt-lg-0" >
          <li>
              <Link to="/">Home</Link>
            </li>
          </ul>
          <ul className="navbar-nav ml-auto">
            <li className="nav-item">
              <div className="search-input">
                <span className="fa fa-search search" id="search" />
                <input
                  type="text"
                  className="form-control"
                  placeholder="Search"
                />
                <div className="dropdown">
                  <button
                    className="btn btn-secondary dropdown-toggle"
                    type="button"
                    data-toggle="dropdown"
                  >
                    <span className="caret" />
                  </button>
                  <ul className="dropdown-menu">
                    <input
                      className="form-control"
                      id="myInput"
                      type="text"
                      placeholder="Filter.."
                    />
                    <li className="pl">
                      <a href="/" />
                      Author
                    </li>
                    <li className="pl">
                      <a href="/" />
                      Title
                    </li>
                    <li className="pl">
                      <a href="/" />
                      Tag
                    </li>
                  </ul>
                </div>
              </div>
            </li>
            {authUser()
              ? (
                <React.Fragment>
                  <li className='login-li-nav'>
                    <Link to="/profile">Profile</Link>
                  </li>
                <li id="login-nav" className='login-li-nav'>
                <a onClick={
                  () => localStorage.removeItem('user')
                  } href="/">Logout</a>
              </li>
              </React.Fragment>
              ) : (
                <React.Fragment>
                  <li id="login-nav" className='login-li-nav'>
                    <a data-toggle="modal" data-target="#loginModal">Login</a>
                  </li>
                  <li className="nav-item">
                    <button
                      type="button"
                      className="btn btn-warning"
                      data-toggle="modal"
                      data-target="#register_modal"
                    >
                      Register
                    </button>
                  </li>
                </React.Fragment>
              )}
          </ul>
        </div>
        <LoginForm />
      </nav>


    );
  }
}

export default Navbar;
