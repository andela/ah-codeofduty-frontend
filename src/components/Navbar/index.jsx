import React, { Component } from 'react';
import 'bootstrap/dist/js/bootstrap.min';
import 'bootstrap/dist/css/bootstrap.min.css';
import './style.scss';
import '../../App.scss';

class Navbar extends Component {
  render() {
    return (
      <nav
        className="navbar navbar-expand-lg navbar-light bg-light font-raleway"
        id="navbar_home"
      >
        <a
          className="logo pl-1 pr-4 mr-3 text-large font-weight-bold App-Logo"
          href="/"
        >
          Authors' Haven
        </a>
        <button
          className="navbar-toggler"
          type="button"
          data-toggle="collapse"
          data-target="#navbarTogglerDemo02"
          aria-controls="navbarTogglerDemo02"
          aria-expanded="false"
          aria-label="Toggle navigation"
        >
          <span className="navbar-toggler-icon" />
        </button>

        <div className="collapse navbar-collapse" id="navbarTogglerDemo02">
          <ul className="navbar-nav mr-auto mt-2 mt-lg-0" />
          <ul className="navbar-nav ml-auto">
            <li className="nav-item">
              <div className="search-input">
                <span className="fa fa-search search" id="search" />
                <input
                  type="text"
                  className="form-control"
                  placeholder="Search"
                />
                <div className="dropdown">
                  <button
                    className="btn btn-secondary dropdown-toggle"
                    type="button"
                    data-toggle="dropdown"
                  >
                    <span className="caret" />
                  </button>
                  <ul className="dropdown-menu">
                    <input
                      className="form-control"
                      id="myInput"
                      type="text"
                      placeholder="Filter.."
                    />
                    <li className="pl">
                      <a href="/" />
                      Author
                    </li>
                    <li className="pl">
                      <a href="/" />
                      Title
                    </li>
                    <li className="pl">
                      <a href="/" />
                      Tag
                    </li>
                  </ul>
                </div>
              </div>
            </li>
            <li className="nav-item">
              <a
                href="/"
                className="nav-link"
                data-toggle="modal"
                data-target="#signup_modal"
              >
                Login &nbsp;
              </a>
            </li>
            <li className="nav-item">
              <button
                type="button"
                className="btn btn-warning"
                data-toggle="modal"
                data-target="#register_modal"
              >
                Register
              </button>
            </li>
          </ul>
        </div>
      </nav>
    );
  }
}

export default Navbar;
