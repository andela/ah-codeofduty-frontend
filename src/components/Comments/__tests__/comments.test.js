import React from 'react';
import { shallow } from 'enzyme';
import { Navbar } from '../../Navbar/Navbar';
import Comment from '../index';

describe('Average Component', () => {
  it('should render Average without throwing an error', () => {
    const wrapper = shallow(<Comment />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Navbar should render correctly', () => {
    const wrapper = shallow(<Navbar />);
    expect(wrapper).toMatchSnapshot();
  });
});
