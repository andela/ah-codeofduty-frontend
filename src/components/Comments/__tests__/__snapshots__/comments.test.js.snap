// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Average Component should render Average without throwing an error 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MyComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <DisqusThread
        id="e94d73ff-fd92-467d-b643-c86889f4b8be"
        path="/article/"
        title="Author's Haven comments"
      />,
      "className": "container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "id": "e94d73ff-fd92-467d-b643-c86889f4b8be",
        "path": "/article/",
        "title": "Author's Haven comments",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <DisqusThread
          id="e94d73ff-fd92-467d-b643-c86889f4b8be"
          path="/article/"
          title="Author's Haven comments"
        />,
        "className": "container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "id": "e94d73ff-fd92-467d-b643-c86889f4b8be",
          "path": "/article/",
          "title": "Author's Haven comments",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
